lib/http/base.js

```diff
@@ -29,42 +32,56 @@ function HttpClient(connectionConfig) {
  * @returns {Object} an object representing the request that was issued.
  */
 HttpClient.prototype.request = function (options) {
-  let request;
+  Logger.getInstance().trace('Request%s - preparing for sending.', requestUtil.describeRequestFromOptions(options));
+
+  let requestPromise;
   const requestOptions = prepareRequestOptions.call(this, options);
   let sendRequest = async function sendRequest() {
-    request = axios.request(requestOptions).then(response => {
+    Logger.getInstance().trace('Request%s - sending.', requestUtil.describeRequestFromOptions(requestOptions));
+    const timer = new ExecutionTimer().start();
+    requestPromise = axios.request(requestOptions).then(response => {
+      const httpResponseTime = timer.getDuration();
+      Logger.getInstance().debug('Request%s - response received after %s milliseconds with status %s.', requestUtil.describeRequestFromOptions(requestOptions), httpResponseTime, response.status);
       sanitizeAxiosResponse(response);
```

Comment:
I believe that to contain parity with the 'requestAsync' method, next to the 'sanitizeAxiosResponse' method, there should be a 'parseResponseData(response)' statement.

However at this point, it may cause a BCR, since JSON data was not parsed previously in this Synch method.